TTHAnalysis/python/plotter/susy-sos/sos_plots.py

```diff
@@ -114,15 +121,125 @@ def procs(GO,mylist):
+        f.write( 'do CARDS="${CARDS} `find  $f -regex .*txt`"\n'  )
+        f.write( 'done\n' )
+        f.write( 'echo ${CARDS}\n' )
+        f.write( 'cd %s\n'%(HIGGSCOMBINEDIR) )
+        f.write( 'eval `scramv1 runtime -sh`\n' )
+        f.write( 'cd -\n' )
+        f.write( '[ -d %s/combinedCards ] || mkdir %s/combinedCards\n'%(ODIR,ODIR) )
+        f.write( 'combineCards.py -S $CARDS > %s/combinedCards/%s.txt\n' %( ODIR, masses) )
+        f.write( '[ -d %s/limits ] || mkdir %s/limits\n'%(ODIR,ODIR) )
+        f.write( 'combine -M Asymptotic %s/combinedCards/%s.txt -n %s -m %s > %s/limits/%s_limit.txt \n'%(ODIR, masses, masses, masses.split('_')[0], ODIR, masses ) )
+        f.write( 'mv higgsCombine%s.Asymptotic.mH%s.root %s/limits \n'%(masses, masses.split('_')[0], ODIR ) )
+    
+    f.close()
+
 def runIt(GO,name):
-    if args.data: name=name+"_data"
+    if args.data and not args.doWhat == "cards" : name=name+"_data"
     if args.norm: name=name+"_norm"
     if args.unc: name=name+"_unc"
-    print name+"\n"
-    if args.doWhat == "plots":  print submit.format(command=' '.join(['python mcPlots.py',"--pdir %s/%s/%s"%(ODIR,YEAR,name),GO,' '.join(['--sP %s'%p for p in (args.inPlots.split(",") if args.inPlots is not None else []) ]),' '.join(['--xP %s'%p for p in (args.exPlots.split(",") if args.exPlots is not None else []) ])]))
-
-    if args.doWhat == "cards":  print submit.format(command=' '.join(['python makeShapeCardsNew.py --savefile',"--outdir %s/%s/%s"%(ODIR,YEAR,name),GO,' '.join(['--sP %s'%p for p in (args.inPlots.split(",") if args.inPlots is not None else []) ]),' '.join(['--xP %s'%p for p in (args.exPlots.split(",") if args.exPlots is not None else []) ]), "--xp='signal(?!.*%s).*'"%args.signalMasses.strip('signal') if args.signalMasses is not None else ''   ]))
-
+    if args.doWhat == "cards": mass = '_'.join(name.split('_')[-2:])
+#    print name.split('_')[-2]
+#    print mass
+#    print name+"\n"
+    if args.doWhat == "plots":  
+        ret = submit.format(command=' '.join(['python mcPlots.py',"--pdir %s/%s/%s"%(ODIR,YEAR,name),GO,' '.join(['--sP %s'%p for p in (args.inPlots.split(",") if args.inPlots is not None else []) ]),' '.join(['--xP %s'%p for p in (args.exPlots.split(",") if args.exPlots is not None else []) ])]))
+
+    if args.doWhat == "cards":  
+        ret = submit.format(command=' '.join(['python makeShapeCardsNew.py --savefile',"--outdir %s/scan/SR/%s/%s/%sfb/TChiWZ/%s/sig_TChiWZ_%s/"%(ODIR,YEAR,name.replace("_%s"%mass,''),LUMI.strip('-l ').replace('.', 'p'),mass,mass),GO,' '.join(['--sP %s'%p for p in (args.inPlots.split(",") if args.inPlots is not None else []) ]),' '.join(['--xP %s'%p for p in (args.exPlots.split(",") if args.exPlots is not None else []) ]), "--xp='signal(?!.*%s).*'"%args.signalMasses.strip('signal') if args.signalMasses is not None else '', "--all-processes", "--asimov=%s"%(args.asimov) if args.asimov is not None else ''   ]))
```

Comment:
I can't understand the inclusion of the --sP and\or --xp flags in this line. What does the card making have to do with selecting or excluding plots?